### DISCUSSION TITLE
Browser File api for nextjs

### DISCUSSION THREAD

**Discussion by QzCurious** [ecyrbe/zodios#263] [Q&A] [answered]
Hi, I'm facing an issue that I can't initialize zodios endpoint if I use zod validation for `File` (`z.instanceof(File)`).
![image](https://user-images.githubusercontent.com/38932402/207563771-d23337f8-f8af-488b-8f33-26e0012e929b.png)

If anyone know how to overcome this? (with following constraint)

- Should support SSR
- [formdata-node](https://www.npmjs.com/package/formdata-node) is not applicable because the type definition is not comply to browser `File`
![image](https://user-images.githubusercontent.com/38932402/207566001-681951a4-f36d-4d0d-af55-2ac2c1855af5.png)

**Comment by ecyrbe** [comment]
I don't think this will work. So maybe use FormData directly like in the example here :
https://www.zodios.org/docs/client#send-multipartform-data-requests
Since FormData is compatible and available in Node 18.
It's been a long time since i did not test this. So maybe you'll also encounter issues with SSR

**Reply by QzCurious** [reply]
I didn't like the fact that `FormData` could not be typed with its fields. And I came up with cleaner solution below (in my opinion). Hope you find it useful as well.

**Comment by QzCurious** [comment, marked as answer]
I've found my problem also asked on a zod issue. Instead of `z.instanceof(File)` which would not be valid on server side, @zanzlender has [a great way](https://github.com/colinhacks/zod/issues/387#issuecomment-1191390673) to validate the input (the check might only be valid on client side).

I'm making it more type safe with 
```ts
z.any() as ZodType<File>
```
And I'm satisfied with it.

**UPDATE!!**
zod recently document the feature for [custom schema](https://github.com/colinhacks/zod#custom-schemas). (seem like this method already being there for a while)

```ts
z.custom<File>()
```
